pkg/engine/diff.go

```diff
@@ -288,7 +288,10 @@ func GetResourceOutputsPropertiesString(
 	for _, k := range keys {
 		out := outs[k]
 
-		// Print this property if it is printable and either ins doesn't have it or it's different.
+		// Print this property if it is printable and if any of the following are true:
+		// - a property with the same key is not present in the inputs
+		// - the property that is present in the inputs is different
+		// - we are doing a refresh, in which case we always want to show state differences
```

Comment:
ah, that does help.  thanks.  i had actually misread it originally and thought it was saying: for refresh, do not print, and that's what really threw me off.